--- ../news_bracch_temp/routes/web.php	2017-03-07 01:49:59.918050700 +0800
+++ routes/web.php	2017-03-03 20:17:38.569307900 +0800
@@ -86,7 +86,58 @@
 
 Route::get('/', 'IndexController@getIndexView');
 
+Route::get('/news_content', function () {
+    return view('site.news.temp.news_content_static');
+});
+
+Route::get('/news_index', function () {
+    return view('site.news.temp.news_index_static');
+});
+
+Route::get('/bookstore_index', function () {
+    return view('site.bookstore.temp.bookstore_index_static');
+});
+
+Route::get('/bookstore_book', function () {
+    return view('site.bookstore.temp.bookstore_book_static');
+});
 
 Route::group(['prefix' => 'admin'], function () {
+
     Voyager::routes();
+
+    // custom routes
+    Route::group(['as' => 'voyager.'], function () {
+        $namespacePrefix = '\\TCG\\Voyager\\Http\\Controllers\\';
+
+        Route::group(['middleware' => ['admin.user']], function () use ($namespacePrefix) {
+            // news-articles Routes
+            Route::group([
+                'as'     => 'news-articles.',
+                'prefix' => 'news-articles',
+            ], function () use ($namespacePrefix) {
+                // Route::get('/', ['uses' => 'NewsArticleController@index', 'as' => 'index']);
+                Route::get('create', ['uses' => 'NewsArticleController@create', 'as' => 'create']);
+                Route::get('{news_article}/edit', ['uses' => 'NewsArticleController@edit', 'as' => 'edit']);
+            });
+
+            // news-posts Routes
+            /*
+            Route::group([
+                'as'     => 'news-posts.',
+                'prefix' => 'news-posts',
+            ], function () use ($namespacePrefix) {
+                // Route::get('/', ['uses' => 'NewsPostController@index', 'as' => 'index']);
+                Route::get('create', ['uses' => 'NewsPostController@create', 'as' => 'create']);
+                Route::get('{news_post}/edit', ['uses' => 'NewsPostController@edit', 'as' => 'edit']);
+                Route::post('/', ['uses' => 'NewsPostController@store', 'as' => 'store']);
+                Route::put('{news_post}   ', ['uses' => 'NewsPostController@update', 'as' => 'update']);
+                Route::patch('{news_post}   ', ['uses' => 'NewsPostController@update', 'as' => 'update']);
+            });
+            */
+            Route::resource('news-posts', 'NewsPostController', ['only' => [
+                'create', 'edit', 'store', 'update'
+            ]]);
+        });
+    });
 });
