--- ../newsposts_bracch_temp/routes/web.php	2017-02-16 23:18:38.258177800 +0800
+++ routes/web.php	2017-03-16 00:26:55.845530500 +0800
@@ -15,7 +15,7 @@
     return view('welcome');
 });
 
-Route::group(['prefix' => 'client'], function () {
+Route::group(['prefix' => 'client', 'as' => 'client.'], function () {
   //Route::get('/login', 'ClientAuth\LoginController@showLoginForm');
   Route::get('/login', 'ClientAuth\LoginController@showLoginForm')->name('login');
   Route::post('/login', 'ClientAuth\LoginController@login');
@@ -28,8 +28,8 @@
 
   Route::post('/password/email', 'ClientAuth\ForgotPasswordController@sendResetLinkEmail');
   Route::post('/password/reset', 'ClientAuth\ResetPasswordController@reset');
-  Route::get('/password/reset', 'ClientAuth\ForgotPasswordController@showLinkRequestForm');
-  Route::get('/password/reset/{token}', 'ClientAuth\ResetPasswordController@showResetForm');
+  Route::get('/password/reset', 'ClientAuth\ForgotPasswordController@showLinkRequestForm')->name('resetRequest');
+  Route::get('/password/reset/{token}', 'ClientAuth\ResetPasswordController@showResetForm')->name('resetPassword');
 });
 
 Route::get('contact', function() {
@@ -86,7 +86,72 @@
 
 Route::get('/', 'IndexController@getIndexView');
 
+Route::get('/news_content', function () {
+    return view('site.news.temp.news_content_static');
+});
+
+Route::get('/news_index', function () {
+    return view('site.news.temp.news_index_static');
+});
+
+Route::get('/bookstore_index', function () {
+    return view('site.bookstore.temp.bookstore_index_static');
+});
+
+Route::get('/bookstore_book', function () {
+    return view('site.bookstore.temp.bookstore_book_static');
+});
+
+Route::get('/login', 'ClientAuth\LoginController@showLoginForm')->name('login');
+Route::post('/login', 'ClientAuth\LoginController@login');
+Route::post('/logout', 'ClientAuth\LoginController@logout')->name('logout');
+
+Route::get('/register', 'ClientAuth\RegisterController@showRegistrationForm')->name('register');
+Route::post('/register', 'ClientAuth\RegisterController@register');
+
+Route::post('/password/email', 'ClientAuth\ForgotPasswordController@sendResetLinkEmail');
+Route::post('/password/reset', 'ClientAuth\ResetPasswordController@reset');
+Route::get('/password/reset', 'ClientAuth\ForgotPasswordController@showLinkRequestForm')->name('resetRequest');
+Route::get('/password/reset/{token}', 'ClientAuth\ResetPasswordController@showResetForm')->name('resetPassword');
 
 Route::group(['prefix' => 'admin'], function () {
+
     Voyager::routes();
+
+    // custom routes
+    Route::group(['as' => 'voyager.'], function () {
+        $namespacePrefix = '\\TCG\\Voyager\\Http\\Controllers\\';
+
+        Route::group(['middleware' => ['admin.user']], function () use ($namespacePrefix) {
+            // news-articles Routes
+            Route::group([
+                'as'     => 'news-articles.',
+                'prefix' => 'news-articles',
+            ], function () use ($namespacePrefix) {
+                // Route::get('/', ['uses' => 'NewsArticleController@index', 'as' => 'index']);
+                Route::get('create', ['uses' => 'NewsArticleController@create', 'as' => 'create']);
+                Route::get('{news_article}/edit', ['uses' => 'NewsArticleController@edit', 'as' => 'edit']);
+            });
+
+            // news-posts Routes
+            /*
+            Route::group([
+                'as'     => 'news-posts.',
+                'prefix' => 'news-posts',
+            ], function () use ($namespacePrefix) {
+                // Route::get('/', ['uses' => 'NewsPostController@index', 'as' => 'index']);
+                Route::get('create', ['uses' => 'NewsPostController@create', 'as' => 'create']);
+                Route::get('{news_post}/edit', ['uses' => 'NewsPostController@edit', 'as' => 'edit']);
+                Route::post('/', ['uses' => 'NewsPostController@store', 'as' => 'store']);
+                Route::put('{news_post}   ', ['uses' => 'NewsPostController@update', 'as' => 'update']);
+                Route::patch('{news_post}   ', ['uses' => 'NewsPostController@update', 'as' => 'update']);
+            });
+            */
+            Route::resource('news-posts', 'NewsPostController', ['only' => [
+                'create', 'edit', 'store', 'update'
+            ]]);
+
+            Route::resource('books', 'BookController');
+        });
+    });
 });
